### ISSUE TITLE
"KBFS path arguments" to `keybase fs` commands under-documented 

### ISSUE THREAD

**Issue by OJFord** [keybase/client#23787]
I couldn't work out how to call e.g. `keybase fs ls` until I found #15547, after having read the help, scoured the website docs, and read the code (as someone who's barely written Go).

The help usage is:
```man
    keybase fs ls [command options] <path>
```

and the error when you don't give a `<path>` is:
> Error parsing command line arguments: ls requires at least one KBFS path argument

A better error would include an explanation of what a 'KBFS path argument' is, i.e. that it must start `/keybase/private` or `/keybase/team` or `/keybase/public/<valid>`.

Perhaps the usage should even be `/keybase/(private|team|public/<group>)[path]` instead of `<path>`.

Further, the 'at least one' in the error message appears to be wrong; supplying two independently valid KBFS path arguments results in the error:
> ERROR file does not exist

I would guess because it's taking it as one path with a space in the middle.

Supplying just `/keybase` gives:
> ERROR RootFS cannot chroot

which could be problematic for using `/keybase/teams`, since I believe that sub-directory doesn't exist until one joins a team.

For what it's worth, the `/keybase` seems redundant to me, and what I was guessing at before I found out how to call it was variations on what turned out to be valid, just without that prefix. (And then the full path to the real location, just in case the command didn't do what I thought/was hoping.)

**Comment by jpaugh** [comment]
Same thing happened to me. I was trying to use `/kbfs/` and `K:\`. FWIW, the syntax is intuitive enough _on Linux_, where `/keybase` corresponds to the filesystem location where KBFS is mounted, and tab-completion works.

**Comment by bartenbach** [comment]
Yeah, this is one of the least intuitive things I've seen in a while. It's not clear at all how this works and the error messages are not helpful.

```
% keybase fs ls /
Error parsing command line arguments: ls requires KBFS path arguments
```

What *is* a KBFS path argument?

```
% keybase fs ls /keybase
▶ ERROR EOF
zsh: exit 2     keybase fs ls /keybase
```

I still have no idea what this wants. Where is the root of this filesystem? Just tell me what you want, keybase.

**Comment by bartenbach** [comment]
Update: so apparently after some reading, starting the systemd user unit is not enough. You *then* have to run the `run_keybase` executable. That seems to start it up, at which point it still doesn't work:

![image](https://user-images.githubusercontent.com/572337/157763324-5f539f31-9161-491e-8f71-5f8120fa203d.png)

**Comment by bartenbach** [comment]
I installed keybase on a different operating system on a different machine, and I get a different result, but unfortunately it still doesn't work:
![image](https://user-images.githubusercontent.com/572337/157766986-83a44b78-d43b-4fdb-92c7-1c482b9040eb.png)

At this point I am just tired of troubleshooting. This is one of the most unbelievably obtuse programs I have ever seen.

**Comment by GwynethLlewelyn** [comment]
Oh... so _that's_ what a 'KBFS path argument' _is_.

Well, well, I totally agree that this ought to be better documented. Or, rather, documented. Period. It is not :)

And I actually suspect why it isn't. You see, different operating systems have different ways of mounting the filesystem, so the 'KBFS path argument' will depend on how exactly _your_ filesystem got mounted. I'm just saying that because I've just tested: on macOS, _if_ the Fuse kernel extension is _not_ running (I seem to have a conflict somewhere preventing it to launch properly), then the command-line `keybase fs ls <path>` will _not_ work.

That is counter-intuitive by itself — I _thought_ that the whole point of using `keybase fs ls <path>` was to sidestep any issues during the weird mounting phase (which will often hiccup, or not work at all, or only sporadically...). Apparently, the CLI tool relies on the filesystem being properly mounted, on the 'expected' place for your operating system. Argh!

**Comment by JoNilsson** [comment]
BUMP!

**Comment by GwynethLlewelyn** [comment]
Thanks for that much-needed bump!

Actually, let me correct myself. On macOS, even if the KBFS is _not_ mounted, you nevertheless can do

    keybase fs ls /keybase/public/

and it will still work. Assuming you're logged in, of course!

**Comment by ChicagoGregg** [comment]
I was looking for more information on this today as well.

I ran into the following link that provides some information on this:

https://gist.github.com/scottrigby/9c03b0db6100285d5b032b87fac00b8a

It suggests the following commands. I still agree the documentation for this is limited.

```
$ keybase fs ls /Volumes/Keybase/team/$TEAM/
foo.com.saltpack
$ keybase fs read /Volumes/Keybase/team/$TEAM/foo.com.saltpack | keybase decrypt

```

**Comment by come-raczy** [comment]
> It suggests the following commands. I still agree the documentation for this is limited.
> 
> ```
> $ keybase fs ls /Volumes/Keybase/team/$TEAM/
> foo.com.saltpack
> $ keybase fs read /Volumes/Keybase/team/$TEAM/foo.com.saltpack | keybase decrypt
> ```

That's an interesting suggestion, sadly it seems to be system-specific - on my system, there isn't any `/Volumes`. The gaps in documentation are indeed frustrating. However, since I found the command that tells me the keybase moutdir, I thought I could try:

    > keybase fs ls $(keybase config get -b mountdir)/team/$TEAM
    Error parsing command line arguments: ls requires KBFS path arguments
    ...

Obviously, no luck there. However, this works fine:

    > ls $(keybase config get -b mountdir)/team/$TEAM
    the-team-big-secret.txt

Which makes me think that it's not just a problem with the documentation.

**Comment by OJFord** [comment]
I think keybase was already acquihired practically-abandonware when I opened this issue 5y ago.
